distributed/protocol/cupy.py

```diff
@@ -3,6 +3,28 @@
 """
 import cupy
 from .cuda import cuda_serialize, cuda_deserialize
+from distutils.version import LooseVersion
+
+
+class PatchedDeviceArray(object):
+    # TODO: This class wont be necessary
+    #       once Cupy<7.0 is no longer supported
+    def __init__(self, ary):
+        self.parent = ary
+        self.vsn = LooseVersion(cupy.__version__)
+
+    def __getattr__(self, name):
+        if name == "parent":
+            raise AttributeError()
+        if self.vsn >= "7.0.0" or name != "__cuda_array_interface__":
+            return getattr(self.parent, name)
+        else:
+            # Cupy<7.0 cannot handle
+            # __cuda_array_interface__['strides'] == None
+            rtn = self.parent.__cuda_array_interface__
+            if rtn.get("strides") is None:
+                rtn.pop("strides")
+            return rtn
```

Comment:
> Is it possible to modify the __cuda_array_interface__ attribute of a cupy array directly..

I do not believe so, the `__cuda_array_interface__` property is immutable.

> What attributes does cupy.asarray need?

I'm sure we could do that instead, but is there a good reason to reimplement everything when we can just toss out one problematic `__cuda_array_interface__` attribute? (note that I might be missunderstanding)

Reply:
Well, what attributes do we actually need here?  I wouldn't be surprised if we *only* needed the `__cuda_array_interface__` attribute, and didn't need to mimic everything else.  If that was the case then I would suggest that we make a simpler object that only exposed that.  It's not a big deal though, and if we're pressed for time I'd be happy to pass on this.

Reply:
Originally I figured forwarding the attrs was a good check in case cupy needed to use any information or code built into the numba array to create the cupy object. However it occurs to me (now a week later of course) that the `__cuda_array_interface__` is sort of "by definition" all that one needs to know, so I suspect that branch of code isn't ever used.

Reply:
> I wouldn't be surprised if we only needed the __cuda_array_interface__ attribute, and didn't need to mimic everything else. If that was the case then I would suggest that we make a simpler object that only exposed that.

Good call - Doing this seems to work fine